app/controllers/home_controller.rb

```diff
@@ -6,7 +6,14 @@ class HomeController < ApplicationController
 
   def index
     @favorite_projects = current_user.favorites
-    @projects = current_power.projects
+    project_records = current_power.projects
+    @projects = []
+    project_records.each do |p_r|
+      p = p_r.as_json
+      p["can_delete"] = p_r.can_delete?(current_user) ? 1 : 0
```

Comment:
instead of doing this. I think it would be easier to read by doing 
updatable_projects_ids = current_power.updatable_projects.pluck(:id)
  project_records.each do |prj| 
    p = prj.attributes
    p['can_edit'] = updatable_projects_ids.include?(p['id'))
  end

You can calculate if the user can delete the project in javascript. (user with edit privilege should be able to see all the samples.)